cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -226,99 +228,73 @@ public String toString() {
-          * Math.max(0, (2 - _balancingConstraint.resourceBalancePercentage(resource)));
-      // Average utilization for the resource.
-      double avgUtilization = clusterModel.load().expectedUtilizationFor(resource) / _numBrokers;
-      avgUtilizationByResource.put(resource, avgUtilization);
-
-      // Maximum, minimum, and standard deviation utilization for the resource.
-      double hottestBrokerUtilization = 0.0;
-      double coldestBrokerUtilization = Double.MAX_VALUE;
-      double variance = 0.0;
-      int numBalancedBrokers = 0;
+      double balanceUpperThreshold = _balancingConstraint.resourceBalancePercentage(resource);
+      double balanceLowerThreshold = Math.max(0, (2 - _balancingConstraint.resourceBalancePercentage(resource)));
+      List<Double> resourceUtilPcts = new ArrayList<>(numBrokers);
+      int i = 0;
       for (Broker broker : clusterModel.brokers()) {
-        double utilization = resource.isHostResource() ?
-            broker.host().load().expectedUtilizationFor(resource) : broker.load().expectedUtilizationFor(resource);
-        double balanceUpperLimit = resource.isHostResource() ?
-            broker.host().capacityFor(resource) * balanceUpperThreshold : broker.capacityFor(resource) * balanceUpperThreshold;
-        double balanceLowerLimit = resource.isHostResource() ?
-            broker.host().capacityFor(resource) * balanceLowerThreshold : broker.capacityFor(resource) * balanceLowerThreshold;
-        if (utilization <= balanceUpperLimit && utilization >= balanceLowerLimit) {
-          numBalancedBrokers++;
-        }
-        double brokerUtilization = broker.load().expectedUtilizationFor(resource);
-        hottestBrokerUtilization = Math.max(hottestBrokerUtilization, brokerUtilization);
-        coldestBrokerUtilization = Math.min(coldestBrokerUtilization, brokerUtilization);
-        variance += (Math.pow(brokerUtilization - avgUtilization, 2) / _numBrokers);
+        double utilization = resource.isHostResource() ? broker.host().load().expectedUtilizationFor(resource)
+                                                       : broker.load().expectedUtilizationFor(resource);
+        double capacity = resource.isHostResource() ? broker.host().capacityFor(resource)
+                                                    : broker.capacityFor(resource);
+        resourceUtilPcts.add(i++, utilization / capacity);
       }
-      _numBalancedBrokersByResource.put(resource, numBalancedBrokers);
-      maxUtilizationByResource.put(resource, hottestBrokerUtilization);
-      minUtilizationByResource.put(resource, coldestBrokerUtilization);
-      stDevUtilizationByResource.put(resource, Math.sqrt(variance));
+      _numBalancedBrokersByResource.put(resource,
+                                        (int) resourceUtilPcts.stream().filter(pct -> pct <= balanceUpperThreshold && pct >= balanceLowerThreshold).count());
```

Comment:
`balanceUpperThreshold` is a value such as `1.1`. It indicates the maximum allowed extent of unbalance for resource utilization. For example, 1.10 means the highest resource usage of a broker should not be above 1.10x of **average** resource utilization of all the brokers.